core/src/main/codegen/templates/Parser.jj

```diff
@@ -3626,6 +3626,11 @@ SqlNode Expression3(ExprContext exprContext) :
 <#list parser.preAtomicRowMethods as method>
     e = ${method} { return e; }
 |
+</#list>
+<#list parser.alternativeTypeConversionLiteralOrIdentifier as method>
```

Comment:
It seems a little weird to have a list of methods (alternativeTypeConversionLiteralOrIdentifier) that has the same name as the only method in the list (AlternativeTypeConversionLiteralOrIdentifier). It seems like AlternativeTypeConversionLiteralOrIdentifier might be able to fit into a more general list of methods.

Could we instead add it to preAtomicRowMethods? It looks like the difference is the syntactic lookahead being added here. But maybe we could just add that for all preAtomicRowMethods?

Reply:
Yeah I think it can just go in preAtomicRowMethods, and adding the syntactic lookahead shouldn't hurt the other methods in that array